firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLitePersistence.java

```diff
@@ -455,6 +459,122 @@ private Cursor startQuery() {
     }
   }
 
+  /**
+   * Encapsulates a query whose parameter list is so long that it might exceed SQLite limit.
+   *
+   * <p>SQLite limits maximum number of host parameters to 999 (see
+   * https://www.sqlite.org/limits.html). This class wraps most of the messy details of splitting a
+   * large query into several smaller ones.
+   */
+  static class LongQuery {
```

Comment:
This is my attempt at refactoring out the common part between two methods that have a long `IN` clause. Let me know if you think it's overengineered.